### PR TITLE
Show "other" option

### PR DESCRIPTION
**Pull request by jeanpimentel**

### CODE REVIEW

File: app/src/main/java/com/ciandt/dragonfly/example/data/local/entities/VersionEntity.kt [hunk 1]
```diff
@@ -40,6 +40,9 @@ data class VersionEntity(
         @ColumnInfo(name = "output_display_names")
         var outputDisplayNames: String = "",
 
+        @ColumnInfo(name = "closed_set")
+        var closedSet: String = "",
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/data/local/entities/VersionEntity.kt:44]: Shoudln't this be a boolean instead?

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 2]
```diff
@@ -81,9 +91,11 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
                 saveImageToGalleryInteractor,
                 benchmarkInteractor)
 
-        presenter.attachView(this)
-        presenter.setUserFeedback(userFeedback)
-        presenter.setClassifications(classifications)
+        presenter.apply {
+            attachView(this@FeedbackActivity)
+            setUserFeedbacks(userFeedbacks)
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: setUserFeedback (feedback is an uncountable noun: https://www.quora.com/What-is-the-plural-of-feedback-Can-you-ever-use-feedbacks)

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 3]
```diff
@@ -214,6 +230,22 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
     }
 
     private fun setupNegativeFeedbackView() {
+        negativeFormCancelButton.setOnClickListener {
+            hideNegativeForm(true)
+        }
+
+        negativeFormConfirmButton.setOnClickListener {
+            hideNegativeForm(false)
+            val selecteds = ArrayList<String>()
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: selectedItems

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 4]
```diff
@@ -237,24 +269,15 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
         })
 
         cancelButton.setOnClickListener {
-            hideNegativeForm()
-        }
-
-        confirmButton.setOnClickListener {
-            val selected = formChipsViews.getSelectedItems().firstOrNull()
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt:244]: selectedItems

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 5]
```diff
@@ -237,24 +269,15 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
...
+            hideNegativeFormInput()
         }
     }
 
     override fun onResume() {
         super.onResume()
         presenter.apply {
             attachView(this@FeedbackActivity)
-            setUserFeedback(userFeedback)
+            setUserFeedbacks(userFeedbacks)
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: userFeedback

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 6]
```diff
@@ -268,11 +291,12 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
         super.onSaveInstanceState(outState)
 
         outState?.apply {
-            putParcelable(USER_FEEDBACK, userFeedback)
+            putParcelableArrayList(USER_FEEDBACKS, userFeedbacks)
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: USER_FEEDBACK

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 7]
```diff
@@ -285,75 +309,128 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
...
             val chips = ArrayList<FeedbackChip>()
             entry.value.mapTo(chips, { FeedbackChip(it) })
 
             val classificationsView = ClassificationsView(this)
+            classificationsView.tag = index
             classificationsView.setTitle(entry.key)
             classificationsView.setChips(chips)
-            classificationsView.select(chips.first())
+
+            var select: FeedbackChip = chips.first()
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: selected or toSelect?

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 8]
```diff
@@ -285,75 +309,128 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
...
+        (0 until otherPredictionsContainer.childCount)
+                .map { otherPredictionsContainer.getChildAt(it) as? ClassificationsView }
+                .forEach {
+                    it?.setSelectable(false)
+                    if (reset) {
+                        it?.deselectAll()
+                        it?.select(0)
+
+                        if (model.closedSet[it?.tag as Int].toBoolean()) {
+                            it.removeChip(it.getChips().size - 1)
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: Won't this remove items that aren't "custom"?

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt [hunk 9]
```diff
@@ -285,75 +309,128 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
...
+                }
+
+        disableNegativeFormConfirm()
+        negativeFormButtonsContainer.visibility = View.GONE
+        toggleContainer.visibility = View.VISIBLE
     }
 
-    override fun setUserFeedback(feedback: Feedback) {
-        userFeedback = feedback
+    override fun setUserFeedbacks(feedbacks: List<Feedback>) {
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: "feedback"(uncountable)

**Review by ivamluz — CHANGES_REQUESTED** [review]
In general, where it says ˜feedbacks˜, think we should us ˜feedbackList" instead. The reason is described below (feedback is an uncountable noun)

File: app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt (hunk 8 above)
Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt]: Changed

File: app/src/main/java/com/ciandt/dragonfly/example/data/local/entities/VersionEntity.kt (hunk 1 above)
Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/data/local/entities/VersionEntity.kt:44]: SQLite does not have a boolean type. We could use integer if it was a single value, but this is a comma-separated value.

Comment (ivamluz) [reply app/src/main/java/com/ciandt/dragonfly/example/data/local/entities/VersionEntity.kt:44]: Got it.

**Pull request merged**